File: Longest Common Prefix 14/Solution.java
public class Solution {
    public static String longestCommonPrefix(String[] strs){
        if(strs == null || strs.length == 0){
            return "";
        }

        int min_length = Integer.MAX_VALUE;
        for(String s: strs) {
            if (s.length() < min_length) {
                min_length = s.length();
            }
        }

        int i = 0; 
        while(i < min_length) {
            for(String s: strs) {
                if (s.charAt(i) != strs[0].charAt(i)) {
                    return strs[0].substring(0,i);
                }
            }
            i++;
        }

        return strs[0].substring(0,i);
    }
    public static void main(String[] args) {
        // Solution s = new Solution();
        String[] q = {"flowz","flow","flowht"};

        System.out.println(longestCommonPrefix(q));

    }
}

File: Longest Common Prefix 14/Solution.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        min_length = float('inf')
        
        for s in strs:
            if len(s) < min_length:
                min_length = len(s)
                
        i = 0
        while i < min_length:
            for s in strs:
                if s[i] != strs[0][i]:
                    return s[:i]
                
            i += 1
        return strs[0][:i]
    
sol = Solution()
q = ["flower","flow","flight"]
print(sol.longestCommonPrefix(q))
print(sol.longestCommonPrefix(["dog","racecar","car"]))